
diff --git a/verilog/dv/ascon/ascon.c b/verilog/dv/ascon/ascon.c
index af8da0a..3e4698c 100644
--- a/verilog/dv/ascon/ascon.c
+++ b/verilog/dv/ascon/ascon.c
@@ -27,5 +27,8 @@ void main()
 	/* Apply configuration */
 	reg_mprj_xfer = 1;
 	while (reg_mprj_xfer == 1);
+
+	// Flag start of the test
+	reg_mprj_datal = 0xB4000000;
 }
 
diff --git a/verilog/dv/ascon/ascon_tb.v b/verilog/dv/ascon/ascon_tb.v
index c9d5777..482f142 100755
--- a/verilog/dv/ascon/ascon_tb.v
+++ b/verilog/dv/ascon/ascon_tb.v
@@ -41,7 +41,10 @@ module ascon_tb;
     wire   tagxSO;
     wire   ascon_readyxSO;
     integer check_time;
+   wire [7:0] checkbits;
 
+   assign checkbits = mprj_io[31:24];
+   
 
 	assign mprj_io[3] = (CSB == 1'b1) ? 1'b1 : 1'bz;
 
@@ -70,7 +73,7 @@ module ascon_tb;
 	initial begin
 		repeat (100) begin
 			repeat (1000) @(posedge clk);
-				// $display("+1000 cycles");
+				$display("+1000 cycles");
 		end
 	
 		$display("%c[1;31m",27);
@@ -84,6 +87,12 @@ module ascon_tb;
 		end 
 	always #(PERIOD) clk = ~clk;
 
+      //Monitor
+   initial begin
+      wait(checkbits == 8'hB4);
+      $display("Monitor: Test MPRJ (RTL) Started!");
+   end
+
 	task write;
     input [max-1:0] rd, i, key, nonce, ass_data, ct; 
     begin
@@ -215,7 +224,7 @@ module ascon_tb;
 		power4 <= 1'b1;
 	end
 
-	always @(mprj_io) begin
+	always @(mprj_io[19:17]) begin
 		#1 $display("MPRJ-IO state = %b ", mprj_io[19:17]);
 	end
 
diff --git a/verilog/dv/cocotb/design_info.yaml b/verilog/dv/cocotb/design_info.yaml
index 39f9875..8c5f3a1 100644
--- a/verilog/dv/cocotb/design_info.yaml
+++ b/verilog/dv/cocotb/design_info.yaml
@@ -1,8 +1,8 @@
-CARAVEL_ROOT: /home/soysiro/vku/ascon_vku/caravel
-MCW_ROOT: /home/soysiro/vku/ascon_vku/mgmt_core_wrapper
+CARAVEL_ROOT: /home/admin/projects/ascon_vku/caravel
+MCW_ROOT: /home/admin/projects/ascon_vku/mgmt_core_wrapper
 PDK: sky130A
-PDK_ROOT: /home/soysiro/vku/ascon_vku/dependencies/pdks
-USER_PROJECT_ROOT: /home/soysiro/vku/ascon_vku
+PDK_ROOT: /home/admin/projects/ascon_vku/dependencies/pdks
+USER_PROJECT_ROOT: /home/admin/projects/ascon_vku
 caravan: false
 clk: 25
 emailto:
diff --git a/verilog/rtl/hs_ascon/ascon_wrapper.v b/verilog/rtl/hs_ascon/ascon_wrapper.v
index f9ae0d4..379f1c7 100644
--- a/verilog/rtl/hs_ascon/ascon_wrapper.v
+++ b/verilog/rtl/hs_ascon/ascon_wrapper.v
@@ -8,14 +8,11 @@ module ascon_wrapper (
     input      clk,
     input      rst,
     input      [5:0] io_in,
-    output reg [2:0] io_out,
-    output reg [10:0] io_oeb
+    output [2:0] io_out,
+    output [10:0] io_oeb
 );
 
-    // Đặt giá trị mặc định cho io_oeb trong khối always
-    always @(*) begin
-        io_oeb = 11'b000_1111_1111;
-    end
+   assign io_oeb = 11'b000_1111_1111;
 
     // Module Ascon
     wire output_dataxSO;
@@ -36,16 +33,9 @@ module ascon_wrapper (
         .ascon_readyxSO(ascon_readyxSO)
     );
 
-    // Gán giá trị cho io_out trong khối always
-    always @(posedge clk or posedge rst) begin
-        if (rst) begin
-            io_out <= 3'b000;
-        end else begin
-            io_out[2] <= output_dataxSO;
-            io_out[1] <= tagxSO;
-            io_out[0] <= ascon_readyxSO;
-        end
-    end
+   assign io_out[2] = output_dataxSO;
+   assign io_out[1] = tagxSO;
+   assign io_out[0] = ascon_readyxSO;
 
 endmodule
 
diff --git a/verilog/rtl/user_project_wrapper.v b/verilog/rtl/user_project_wrapper.v
index e7aa901..03cb306 100644
--- a/verilog/rtl/user_project_wrapper.v
+++ b/verilog/rtl/user_project_wrapper.v
@@ -92,7 +92,7 @@ module user_project_wrapper #(
     .rst(io_in[15]),
     .io_in(io_in[14:9]),
     .io_out(io_out[19:17]),
-    .io_oeb({io_oeb[16:9], io_oeb[19:17]})
+    .io_oeb(io_oeb[19:9])
 );
 
 
